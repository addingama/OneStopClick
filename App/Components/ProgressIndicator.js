import React, { Component, PropTypes } from 'react'
import { View, Text, ActivityIndicator } from 'react-native'
import I18n from 'react-native-i18n'
import HideableView from 'react-native-hideable-view'
import styles from './Styles/ProgressIndicatorStyle'

export default class ProgressIndicator extends Component {
  // Prop type warnings
  static propTypes = {
    show: PropTypes.bool.isRequired,
    text: PropTypes.string.isRequired,
  }
  
  // Defaults for props
  static defaultProps = {
    show: false,
    text: I18n.t('loading')
  }

  render () {
    const { show, text } = this.props
    return (
      <HideableView style={styles.messageBox} visible={show}>
        <ActivityIndicator size='large' color={'#ffffff'}/>
        <Text style={styles.messageText}>{text}</Text>
      </HideableView>
    )
  }
}
