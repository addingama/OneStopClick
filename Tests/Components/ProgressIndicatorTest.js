import 'react-native'
import React from 'react'
import ProgressIndicator from '../../App/Components/ProgressIndicator'
import renderer from 'react-test-renderer'

test('ProgressIndicator component renders correctly if show is true', () => {
  const tree = renderer.create(<ProgressIndicator show text='howdy' />).toJSON()
  expect(tree).toMatchSnapshot()
})

test('ProgressIndicator component does not render if show is false', () => {
  const tree = renderer.create(<ProgressIndicator title='howdy' show={false} />).toJSON()
  expect(tree).toMatchSnapshot()
})

test('ProgressIndicator component renders correctly if title prop is set', () => {
  const tree = renderer.create(<ProgressIndicator title='howdy' show />).toJSON()
  expect(tree).toMatchSnapshot()
})
