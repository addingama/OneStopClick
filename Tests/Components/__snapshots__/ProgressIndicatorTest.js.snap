// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgressIndicator component does not render if show is false 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "alignSelf": "center",
      "backgroundColor": "rgba(52, 52, 52, 0.8)",
      "borderRadius": 10,
      "flex": 1,
      "justifyContent": "center",
      "marginLeft": 20,
      "marginRight": 20,
      "opacity": 0,
      "padding": 10,
      "position": "absolute",
    }
  }
>
  <ActivityIndicator
    animating={true}
    color="#ffffff"
    hidesWhenStopped={true}
    size="large"
  />
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "white",
        "fontSize": 12,
        "paddingTop": 10,
        "textAlign": "center",
      }
    }
  >
    Please wait...
  </Text>
</View>
`;

exports[`ProgressIndicator component renders correctly if show is true 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "alignSelf": "center",
      "backgroundColor": "rgba(52, 52, 52, 0.8)",
      "borderRadius": 10,
      "flex": 1,
      "justifyContent": "center",
      "marginLeft": 20,
      "marginRight": 20,
      "opacity": 1,
      "padding": 10,
      "position": "absolute",
    }
  }
>
  <ActivityIndicator
    animating={true}
    color="#ffffff"
    hidesWhenStopped={true}
    size="large"
  />
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "white",
        "fontSize": 12,
        "paddingTop": 10,
        "textAlign": "center",
      }
    }
  >
    howdy
  </Text>
</View>
`;

exports[`ProgressIndicator component renders correctly if title prop is set 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "alignSelf": "center",
      "backgroundColor": "rgba(52, 52, 52, 0.8)",
      "borderRadius": 10,
      "flex": 1,
      "justifyContent": "center",
      "marginLeft": 20,
      "marginRight": 20,
      "opacity": 1,
      "padding": 10,
      "position": "absolute",
    }
  }
>
  <ActivityIndicator
    animating={true}
    color="#ffffff"
    hidesWhenStopped={true}
    size="large"
  />
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "white",
        "fontSize": 12,
        "paddingTop": 10,
        "textAlign": "center",
      }
    }
  >
    Please wait...
  </Text>
</View>
`;
